homeassistant/components/somfy/config_flow.py

```diff
@@ -0,0 +1,146 @@
+"""Config flow for Somfy."""
+import asyncio
+import logging
+
+import async_timeout
+
+from homeassistant import config_entries
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.core import callback
+from .const import CLIENT_ID, CLIENT_SECRET, DOMAIN
+
+AUTH_CALLBACK_PATH = '/auth/somfy/callback'
+AUTH_CALLBACK_NAME = 'auth:somfy:callback'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def register_flow_implementation(hass, client_id, client_secret):
+    """Register a flow implementation.
+
+    client_id: Client id.
+    client_secret: Client secret.
+    """
+    hass.data[DOMAIN][CLIENT_ID] = client_id
+    hass.data[DOMAIN][CLIENT_SECRET] = client_secret
+
+
+@config_entries.HANDLERS.register('somfy')
+class SomfyFlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Instantiate config flow."""
+        self.code = None
+
+    async def async_step_import(self, user_input=None):
+        """Handle external yaml configuration."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason='already_setup')
+        return await self.async_step_auth()
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason='already_setup')
+
+        return await self.async_step_auth()
+
+    async def async_step_auth(self, user_input=None):
+        """Create an entry for auth."""
+        # Flow has been triggered from Somfy website
+        if user_input:
+            return await self.async_step_code(user_input)
+
+        try:
+            with async_timeout.timeout(10):
+                url, _ = await self._get_authorization_url()
+        except asyncio.TimeoutError:
+            return self.async_abort(reason='authorize_url_timeout')
+        except Exception:  # pylint: disable=broad-except
```

Comment:
Can you be more specific in the exception that are getting raised ?

Reply:
I've removed this catch them all. I did exactly like the Nest component. But I double check and cannot guess what can be raised except the TimeoutError.